crates/sui-types/src/crypto.rs

```diff
@@ -1527,13 +1532,23 @@ where
     }
 }
 
-pub fn sha3_hash<S: Signable<Sha3_256>>(signable: &S) -> [u8; 32] {
-    let mut digest = Sha3_256::default();
+fn hash<S: Signable<H>, H: HashFunction<DIGEST_SIZE>, const DIGEST_SIZE: usize>(
```

Comment:
for hashes we do still use it (prefix with type tag) but not for signing. since we discussed object id computation should use intent for hashing, i can address both. 

we can have another enum called `HashIntent`?

Reply:
yes @joyqvq HashIntent is fine